### ISSUE TITLE
dockerfile_template_conda.jinja2 mismatch between git repo and built docker image

### ISSUE THREAD

**Issue by gpelouze** [QCDIS/NaaVRE#1216]
When building [`qcdis/n-a-a-vre:v2.4.0-alpha.1`](https://hub.docker.com/layers/qcdis/n-a-a-vre/v2.4.0-alpha.1/images/sha256-8d8f266125e64bf285d41d42f3f209a6dc2ca972c237abe40f4b8b7907c5a8a0?context=explore) from https://github.com/QCDIS/NaaVRE/tree/v2.4.0-alpha.1, we don’t get the expected `jupyterlab_vre/templates/dockerfile_template_conda.jinja2`.

* In the repo ([`jupyterlab_vre/templates/dockerfile_template_conda.jinja2#L6`](https://github.com/QCDIS/NaaVRE/blob/v2.4.0-alpha.1/jupyterlab_vre/templates/dockerfile_template_conda.jinja2#L6)):

```Dockerfile
[...]
RUN conda-pack -p /opt/conda/envs/venv -o /tmp/env.tar && \
    mkdir /venv && cd /venv && tar xf /tmp/env.tar && \
    rm /tmp/env.tar
[...]
```

* In the docker image:

```shell
$ docker run -it --rm qcdis/n-a-a-vre:v2.4.0-alpha.1 cat /venv/lib/python3.11/site-packages/jupyterlab_vre/templates/dockerfile_template_conda.jinja2
[...]
RUN conda-pack -p /venv -o /tmp/env.tar && \
    mkdir /venv && cd /venv && tar xf /tmp/env.tar && \
    rm /tmp/env.tar
[...]
```

As a result, NaaVRE generates cells that can’t be built. Example ([cell](https://github.com/QCDIS/NaaVRE-cells-test-2/blob/c8aeb32844bca2e4cc0b9a4f3c3dbba1ccf92388/retrieve-bud-burst-data-from-minio-gabriel-pelouze-at-lifewatch-eu/Dockerfile#L6) and [action](https://github.com/QCDIS/NaaVRE-cells-test-2/actions/runs/8094088838/job/22118052350)):

```
> [build 4/5] RUN conda-pack -p /venv -o /tmp/env.tar &&     mkdir /venv && cd /venv && tar xf /tmp/env.tar &&     rm /tmp/env.tar:
0.335 Collecting packages...
0.335 CondaPackError: Environment path '/venv' doesn't exist
```

---

The artifact [`jupyterlab_vre-0.1.0-py3-none-any.whl`](https://github.com/QCDIS/NaaVRE/actions/runs/8067917609/artifacts/1279521689) generated during the build contains the same template file as the repo.

---

I can reproduce the problem locally from branch `1160-refactor-cell-dockerfile-template`:

- `make release` generates a wheel file with the expected template
- the locally-built docker image contains the wrong template (matching [`build-container.yml` workflow](https://github.com/QCDIS/NaaVRE/blob/9eeb097862fcf5c31ad189382af9a89f168e200c/.github/workflows/build-container.yml#L49)):

```shell
export DOCKER_FOLDER=vanilla
cp docker/init_script.sh docker/$DOCKER_FOLDER/
cp docker/start-jupyter.sh docker/$DOCKER_FOLDER/
cp docker/start-jupyter.sh docker/$DOCKER_FOLDER/
cp docker/start-jupyter-venv.sh docker/$DOCKER_FOLDER/
cp -r docker/repo_utils docker/$DOCKER_FOLDER/repo_utils
cp environment.yml docker/$DOCKER_FOLDER/
mv docker/$DOCKER_FOLDER/environment.yml docker/$DOCKER_FOLDER/naa-vre-environment.yaml
yq '.dependencies[-1].pip' docker/$DOCKER_FOLDER/naa-vre-environment.yaml 2> /dev/null || cat $f | yq '.dependencies += [{"pip": []}]' docker/$DOCKER_FOLDER/naa-vre-environment.yaml -i
yq '.dependencies[-1].pip += ["/home/jovyan/jupyterlab_vre-0.1.0-py3-none-any.whl"]' docker/$DOCKER_FOLDER/naa-vre-environment.yaml -i
if test -f "docker/$DOCKER_FOLDER/environment.yaml"; then yq 'del(.channels[] | select(. == "defaults"))' docker/$DOCKER_FOLDER/environment.yaml -i; fi
yq 'del(.channels[] | select(. == "defaults"))' docker/$DOCKER_FOLDER/naa-vre-environment.yaml -i
cp dist/jupyterlab_vre-0.1.0-py3-none-any.whl docker/$DOCKER_FOLDER
docker build docker/$DOCKER_FOLDER -t tmp-naavre
```

```shell
$ docker run -it --rm tmp-naavre cat /venv/lib/python3.11/site-packages/jupyterlab_vre/templates/dockerfile_template_conda.jinja2
[...]
RUN conda-pack -p /venv -o /tmp/env.tar && \
    mkdir /venv && cd /venv && tar xf /tmp/env.tar && \
    rm /tmp/env.tar
[...]
```

**Comment by gpelouze** [comment]
The culprit seems to be `conda-pack` at [`docker/vanilla/Dockerfile#L10`](https://github.com/QCDIS/NaaVRE/blob/106ca12bb31b7b10b10dfcbc9441c94c5f2bf130/docker/vanilla/Dockerfile#L10), which moves the newly installed environment from `/opt/conda/envs/venv/` to `/venv`. In the process, it zealously replaces `/opt/conda/envs/venv` with `/venv` in our template.

Building only the `naavre-build` stage:

```shell
$ docker build docker/$DOCKER_FOLDER --target naavre-build -t tmp-naavre-build
$ docker run -it --rm tmp-naavre-build cat /opt/conda/envs/venv/lib/python3.11/site-packages/jupyterlab_vre/templates/dockerfile_template_conda.jinja2
[...]
RUN conda-pack -p /opt/conda/envs/venv -o /tmp/env.tar && \
[...]

$ docker run -it --rm tmp-naavre-build cat /venv/lib/python3.11/site-packages/jupyterlab_vre/templates/dockerfile_template_conda.jinja2
[...]
RUN conda-pack -p /venv -o /tmp/env.tar && \
[...]
```

Which then gets copied to the `runtime` stage:

```shell
$ docker build docker/$DOCKER_FOLDER -t tmp-naavre
$ docker run -it --rm tmp-naavre cat /venv/lib/python3.11/site-packages/jupyterlab_vre/templates/dockerfile_template_conda.jinja2
[...]
RUN conda-pack -p /venv -o /tmp/env.tar && \
[...]
```

**Comment by gpelouze** [comment]
Closed by #1217
